sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -261,9 +272,13 @@ private void exportCurrentBatch() {
         return;
       }
 
+      for (ReadableSpan span : batch) {
+        batchData.add(span.toSpanData());
```

Comment:
For good or ill, moving this down here will make the CPU usage a bit more spiky, since we'll only do this conversion all at once when the queue is full, or when the timer expires. The current implementation will spread out the conversion smoothly, since it happens as each span is added to the queue. Which way is better? I'm not sure, honestly, but it's something to keep in mind.

Reply:
Yeah there's some tradeoff - I guess for a thread with a defined "interval", I'd sort of expect it to wake, work, sleep. We have a bit of a pathological behavior right now where it's fairly conceivable for this thread to be sleeping and waking in a thrashing manner if a span is recorded every 101ms for example - sleeping/waking is very expensive and I suspect we didn't intend that to be possible.